daemon/daemon.go

```diff
@@ -403,8 +395,52 @@ func (d *daemon) Run() error {
 
 	reflection.Register(s)
 	log.Info().Msg("Reflection Registration is called.... ")
-	if err := d.Sleep(); err != nil {
-		log.Warn().Msgf("Error in sleep %v", err)
+
+	// initialize schedulers
+	if err := d.initializeScheduler(); err != nil {
+		return err
 	}
+
 	return s.Serve(lis)
 }
+
+// calculateTotalConsumption will add up all running events resources
+func calculateTotalConsumption(d *daemon) int {
+	var totalConsumption int
+	for _, e := range d.eventPool.GetAllEvents() {
+		config := e.GetConfig()
+		if config.Status == Running {
+			totalConsumption += config.Capacity
+		}
+		// note that this is very raw calculation
+		// suspended resources could be excluded
+		totalConsumption += config.Available + len(e.GetTeams())
+	}
+	return totalConsumption
+}
+
+func (d *daemon) initializeScheduler() error {
```

Comment:
Create a test for this function

Reply:
@gianmarcomennecozzi
I think, it is enough to write test function for RunScheduler function, since initializeScheduler just initiating configuration and calls RunScheduler.